src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs

```diff
@@ -141,6 +141,48 @@ public void CreateDefaultBuilder_EnablesValidateOnBuild()
             Assert.Throws<AggregateException>(() => hostBuilder.Build());
         }
 
+        [Theory]
+        [InlineData("Beta-Testing"), InlineData("Another-Random-Env")]
+        public void UseEnvironmentIsOverwrittenByAdditionalCalls(string environment)
+        {
+            var expectedEnvironment = "SomeOtherEnvironment";
+            using var host = new HostBuilder()
+                .UseEnvironment(environment)
+                .ConfigureHostConfiguration(configBuilder =>
+                {
+                    configBuilder.AddInMemoryCollection(new[]
+                    {
+                        new KeyValuePair<string, string>(
+                            HostDefaults.EnvironmentKey, expectedEnvironment)
+                    });
+                })
+                .Build();
+
+            var hostEnv = host.Services.GetRequiredService<IHostEnvironment>();
+            Assert.Equal(expectedEnvironment, hostEnv.EnvironmentName);
+        }
+
+        [Theory]
+        [InlineData("Beta-Testing"), InlineData("Another-Random-Env")]
+        public void LastCallToUseEnvironmentWins(string environment)
```

Comment:
Would it make sense to add another explicit call to `.UseEnvironment` as well?

Reply:
There is already one call to it. Did you mean to include another, and have two calls?

```diff
                .ConfigureHostConfiguration(configBuilder =>
                {
                    configBuilder.AddInMemoryCollection(new[]
                    {
                        new KeyValuePair<string, string>(
                            HostDefaults.EnvironmentKey, willBeOverwritten)
                    });
                })
+               .UseEnvironment(someOtherEnvironment)
                .UseEnvironment(environment)
                .Build();
```

Reply:
> have two calls

Correct. The name of the test is `LastCallToUseEnvironmentWins`. But the test only calls `UseEnvironment` once. There isn't a "previous" call to it in order to test if the "last call" wins or not.

Reply:
But there is, the:

```csharp
.ConfigureHostConfiguration(configBuilder =>
{
    configBuilder.AddInMemoryCollection(new[]
    {
        new KeyValuePair<string, string>(
            HostDefaults.EnvironmentKey, willBeOverwritten)
    });
})
```

Does the same thing that `UseEnvironment` does, but I'll add another anyway.